extensions/gitpod-web/src/util/port.ts

```diff
@@ -0,0 +1,172 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+/// <reference path='../../../../src/vscode-dts/vscode.d.ts'/>
+/// <reference path='../../../../src/vscode-dts/vscode.proposed.resolvers.d.ts'/>
+
+import * as workspaceInstance from '@gitpod/gitpod-protocol/lib/workspace-instance';
+import { GitpodExtensionContext } from 'gitpod-shared';
+import { PortsStatus, PortAutoExposure, PortVisibility } from '@gitpod/supervisor-api-grpc/lib/status_pb';
+import { TunnelVisiblity, TunnelPortRequest } from '@gitpod/supervisor-api-grpc/lib/port_pb';
+import { URL } from 'url';
+import * as util from 'util';
+import * as vscode from 'vscode';
```

Comment:
@mustard-mh I gave a little more thought about this, let's decouple `ports.ts` from vscode API so you can remove the vscode API mocks, this will also ensure that if someone wants to add more tests in the future they don't have the wrong idea of keep mocking vscode API, if we really need vscode API in our tests in the future then we'll have to look into moving gitpod-web to it's own repo